src/pgduckdb_filter.cpp

```diff
@@ -17,41 +19,49 @@ TemplatedFilterOperation(Datum &value, const duckdb::Value &constant) {
 	return OP::Operation((T)value, constant.GetValueUnsafe<T>());
 }
 
+template <class OP>
+bool
+StringFilterOperation(Datum &value, const duckdb::Value &constant) {
+	const auto ptr = (text*)DatumGetPointer(value);
+	if (ptr == nullptr || constant.IsNull()) {
+		return false; // Comparison to NULL always returns false.
+	}
+
+	text* tunpacked = pg_detoast_datum_packed(ptr);
```

Comment:
This is going to be problematic/buggy  with multiple threads. There is equivalent function in pgduckdb_detoast.cpp - `DetoastPostgresDatum` which should be safe in duckdb execution.

Reply:
Thanks for the review @mkaruza! I got the inspiration for this from the [text_to_cstring](https://github.com/postgres/postgres/blob/28a1121fd91235e09793c9ce64cb5539719afa30/src/backend/utils/adt/varlena.c#L220-L221) PG function, and I got the sense from @JelteF that we cannot use multiple thread in DuckDB execution when touching PG data anyway.

I was also hoping not to have to allocate + copy the PG string if possible.

Happy to use `DetoastPostgresDatum` though.

Reply:
@Y-- we can use multiple threads for heap scan (there is also GUC variable where you can specify max threads that can be spawned for single table scan). FYI, joining two heap tables also runs in separate threads.

Reply:
I see, make sense, thanks! And I assume there's no way to have a no-alloc & no-copy version of `DetoastPostgresDatum` in this context then?